### PR TITLE
Azure Blob Storage as storage backend for Persisted Operations

### PR DESCRIPTION
**Pull request by cimnine**
This PR proposes a new adapter to use Azure Blob Storage as storage backend for (automatically) persisted operations.

At least on Azure, Redis-compatible cache instances are comparably costly and mounting a shared filesystem is not always possible – and often not desired either. The Blob Storage on the other end is almost free.

Also, it's relatively easy to upload persisted operations extracted from a client in a CI/CD pipeline, whereas this is tricky with the existing options.

Please let me know if you're interested in including this as an officially supported storage backend. Otherwise, I will probably publish it independently.

### CODE REVIEW

**Comment by michaelstaib** [comment]
Hey @cimnine thanks for your contribution.

Tell is when you need a review.

**Comment by cimnine** [comment]
@michaelstaib I believe now it would be ready for a first round of scrutiny.

**Comment by michaelstaib** [comment]
I will have a look today

File: src/HotChocolate/PersistedOperations/src/PersistedOperations.AzureBlobStorage/AzureBlobOperationDocumentStorage.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    /// </summary>
+    /// <param name="containerClient">The blob container client instance.</param>
+    /// <param name="blobNamePrefix">This prefix string is prepended before the hash of the document.</param>
+    /// <param name="blobNameSuffix">This suffix is appended after the hash of the document.</param>
+    public AzureBlobOperationDocumentStorage(
+        BlobContainerClient containerClient,
+        string blobNamePrefix,
+        string blobNameSuffix)
+    {
+        ArgumentNullException.ThrowIfNull(containerClient);
```
Comment (michaelstaib) [inline src/HotChocolate/PersistedOperations/src/PersistedOperations.AzureBlobStorage/AzureBlobOperationDocumentStorage.cs:39]: We cannot use `ArgumentNullException.ThrowIfNull` if you want this to be ported back to 14 as 14 still supports .NET Standard 2.0.

**Comment by cimnine** [comment]
@michaelstaib I would appreciate it if this change was mentioned in the release notes. Somehow I can't find it in https://github.com/ChilliCream/graphql-platform/releases/tag/15.0.0-rc.1.1 .

**Pull request merged**